docs/modules/development/pages/build-from-source.adoc

```diff
@@ -0,0 +1,139 @@
+  As an example for PostgreSQL 13.x you need the package `iplike-pgsql13`
+* JRRD2 if you want to use RRDTool instead of the default Java based RRD implementation JRobin
+* RRDtool from the official package repositories
+
+TIP: You can set the environment `JAVA_HOME` in `/etc/environment` if you want additional control about the Java JDK used during builds.
+Changes are applied immediately with `source /etc/environment`.
+
+TIP: Within the source tree we ship a Maven version which is used by default.
+The environment variable `MVN` allows you using your own installed Maven version.
+
+== Get the source code
+
+First you need the source code from the GitHub repository.
+The default branch is `develop` which is the release candidate for the major release.
+You can check out a specific version tag if you want or switch to a branch you want to build from.
+The command `git tag -l` or `git branch -r` gives you available branches or version tags.
+
+.Get the source from GitHub in your home directory
+[source, console]
+----
+git clone https://github.com/OpenNMS/opennms.git ~/dev/opennms
+cd ~/dev/opennms
+----
+
+== Compile and assemble from source
+
+The build is divided into two steps.
+The first step compiles the source code and the second step assembles the compiled artefacts in a way you can run OpenNMS {page-component-title} locally.
+To speed up the build steps, we explicitly disable tests and don't create a source tarball file.
+
+.Compile the source code without executing test tasks and skip the source tarball creation
+[source, console]
+----
+./compile.pl -Dbuild.skip.tarball=true -DskipTests
+----
+
+.Assemble the build artefacts in the target directory
+[source, console]
+----
+./assemble.pl -p dir -DskipTests
+----
+
+The profile argument `-p dir` allows running OpenNMS {page-component-title} directly from the target directory.
+
+NOTE: If you build from source the first time, it takes some time.
+During the first build the Maven dependencies downloads is about ~4GB.
+The Maven repository is by default located in the `$\{HOME}/.m2` directory.
+
+== Run your build locally
+
+The following instructions require the release version.
+The release version is defined in the root `pom.xml` in the source tree and changes with every major release.
+This release version depends on which version tag or branch you have checked out and try to build.
+It is helpful to extract the release version in an environment variable and allows you to repeat the following instructions more easily.
+
+TIP: Instead of maintaining a local PostgreSQL installation you can also run an ephemeral PostgreSQL instance as a container.
+     The command `sudo docker run -d -e POSTGRES_HOST_AUTH_METHOD=trust -p 5432:5432 postgres:13` starts an instance which allows connections without authentication for development.
+     If you want to persist the database on your host for later use, you can add `-v $(pwd)/data:/var/lib/postgresql/data`.
```

Comment:
@patrick-schweizer I've added a tip here, is this enough to resolve your suggestion?